metrics/metrics/src/main/java/io/helidon/metrics/ExponentiallyDecayingReservoir.java

```diff
@@ -49,6 +54,49 @@
     private static final double DEFAULT_ALPHA = 0.015;
     private static final long RESCALE_THRESHOLD = TimeUnit.HOURS.toNanos(1);
 
+    /*
+     * Avoid computing the current time in seconds during every reservoir update by updating its value on a scheduled basis.
+     */
+    private static final long CURRENT_TIME_IN_SECONDS_UPDATE_INTERVAL_MS = 250;
+
+    private static final List<Runnable> CURRENT_TIME_IN_SECONDS_UPDATERS = new ArrayList<>();
+
+    private static final ScheduledExecutorService CURRENT_TIME_UPDATER_EXECUTOR_SERVICE = initCurrentTimeUpdater();
+
+    private static final Logger LOGGER = Logger.getLogger(ExponentiallyDecayingReservoir.class.getName());
+
+    private volatile long currentTimeInSeconds;
+
+    private static ScheduledExecutorService initCurrentTimeUpdater() {
+        ScheduledExecutorService result = Executors.newSingleThreadScheduledExecutor();
+        result.scheduleAtFixedRate(ExponentiallyDecayingReservoir::updateCurrentTimeInSecondsForAllReservoirs,
+                CURRENT_TIME_IN_SECONDS_UPDATE_INTERVAL_MS,
+                CURRENT_TIME_IN_SECONDS_UPDATE_INTERVAL_MS, TimeUnit.MILLISECONDS);
+        return result;
+    }
+
+    static void onServerShutdown() {
+        CURRENT_TIME_UPDATER_EXECUTOR_SERVICE.shutdown();
+        try {
+            boolean stoppedNormally =
+                    CURRENT_TIME_UPDATER_EXECUTOR_SERVICE.awaitTermination(CURRENT_TIME_IN_SECONDS_UPDATE_INTERVAL_MS * 10,
+                            TimeUnit.MILLISECONDS);
+            if (!stoppedNormally) {
+                LOGGER.log(Level.WARNING, "Shutdown of current time updater timed out; continuing");
+            }
+        } catch (InterruptedException e) {
+            LOGGER.log(Level.WARNING, "InterruptedException caught while stopping the current time updater; continuing");
```

Comment:
Nit: if you "eat" an `InterruptedException` you should do `Thread.currentThread().interrupt()` to preserve its interrupted status. Obviously here stuff is shutting down so it probably doesn't matter.